File: week1/pset1/credit.c
#include <stdio.h>
int main(void)
{
    long number;
    printf("Number: ");
    scanf("%ld", &number);
    long n=number;
    int sum1=0;
    int sum2=0;
    while (n>0)
    {
        int digit=n%10;
        sum1=sum1+digit;
        n=n/10;
        if (n>0)
        {
            int digit2=n%10;
            digit2=digit2*2;
            if (digit2>9)
            {
                sum2=sum2+(digit2/10)+(digit2%10);
            }
            else
            {
                sum2=sum2+digit2;
            }
            n=n/10;
        }
    }
    int total=sum1+sum2;
        if (total%10==0)
        {
            printf("VALID\n");
        }
        else
        {
            printf("INVALID\n");
        }
}

File: week6/pset6/credit.py
number = input("Number: ")
sum = 0
alternate = False
for digit in number[::-1]:
    n = int(digit)
    if alternate:
        n *= 2
        if n > 9:
            n -= 9
    sum += n
    alternate = not alternate
if sum % 10 == 0:
    print("VALID")
else:
    print("INVALID")